buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -1631,7 +1631,7 @@ public ByteBuffer nioBuffer(int index, int length) {
         ByteBuffer[] buffers = nioBuffers(index, length);
 
         if (buffers.length == 1) {
-            return buffers[0].duplicate();
+            return buffers[0];
```

Comment:
this looks alright ... I am not sure about the other changes but this looks like the one we want to apply 100 %